packages/gatsby-theme-store/src/sdk/masterData/index.ts

```diff
@@ -0,0 +1,85 @@
+export type DataInput = {
+  [key: string]: unknown
+}
+
+export type MasterDataSaveResponse = {
+  DocumentId: string
+  Href: string
+  Id: string
+}
+
+export type MasterData = (
+  entity: string
+) => {
+  /**
+   * Create a new document
+   */
+  create: (data: DataInput) => Promise<MasterDataSaveResponse>
+  /**
+   * `upsert` query updates a document if it exists, or creates a new.
+   */
+  upsert: (data: DataInput) => Promise<MasterDataSaveResponse>
+  /**
+   * Similar to `upsert`, the `upsertial` query updates partial data of
+   * document if it exists, or creates a new.
+   *
+   * The difference is that data that is not requested for updating will
+   * be kept.
+   */
+  upsertial: (data: DataInput) => Promise<MasterDataSaveResponse>
+  /**
+   * `update` updates an existing document.
+   *
+   * Similar to `upsert`, however it does not create a document does not exist.
+   */
+  update: (id: string, data: DataInput) => Promise<MasterDataSaveResponse>
+  /**
+   * `findUnique` method lets you retrieve a single document by id.
+   */
+  findUnique: (id: string) => Promise<unknown>
+  /**
+   * `delete` method deletes a document by id.
+   */
+  delete: (id: string) => Promise<unknown>
+}
+
+type Method = 'POST' | 'GET' | 'PATCH' | 'DELETE' | 'PUT'
+
+const fetchy = (
+  url: string,
+  method: Method,
+  data?: DataInput
+): Promise<MasterDataSaveResponse> =>
+  new Promise((resolve, reject) => {
+    fetch(url, {
+      headers: {
+        'Content-Type': 'application/json',
+        accept: 'application/vnd.vtex.ds.v10+json',
+      },
+      method,
+      body: JSON.stringify(data),
+    }).then((response) => {
+      if (response.ok) {
+        resolve(response.json())
+      } else {
+        reject(new Error())
+      }
+    })
+  })
+
+const masterDataApi: MasterData = (entity) => {
+  const url = `/api/dataentities/${entity}/documents`
```

Comment:
We can have both, one with just the fetchers and a hook wrapping this API.

We could have two just [like Apollo does](https://www.apollographql.com/docs/react/api/react/hooks/#uselazyquery):
```tsx
const { data, loading, error } = useMasterData({ entity, id, data })
const [fetchMasterdata, { called, loading, data, error }] = useLazyMasterData(entity)
```

Reply:
We could have a hook to wrap fetch APIs like this :point-up:

Reply:
Maybe use swr

Reply:
Carrefour had to add this to their store:
https://github.com/vtex-sites/carrefourbrfood.store/pull/199/files#diff-a6f3904c41211be170d9b0eb87e49ea8d02d0fd5a7224d6772174b425dacf90fR218

Do we need to add this to our packages?